app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,66 @@
+require 'date'
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.order(:due_date, :completion_date)
+  end
+
+  def show
+    @task = Task.find(params[:id])
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id].to_i)
+    redirect_to task_path unless @task
+  end
+
+  def create
+    @task = Task.new(task_params)
+
+    if @task.save
+      flash[:notice] = "Task saved"
```

Comment:
nice use of flash, but you don't end up using this at all in the view!

We'll be going over flash formally next week

Reply:
I was wondering why it wasn't showing...